### ISSUE TITLE
Upgrade: ImportError: cannot import name path

### ISSUE THREAD

**Issue by pengxiao78** [chapmanb/bcbio-nextgen#990]
I am installing bcbio in a cluster computer.

1. I set and unset ENV variables using the following commands

export BCBIO_PATH="/work/[USERGROUP]/[USERNAME]/bcbio" 
export _JAVA_OPTIONS=-Xmx2g
unset PYTHONHOME
unset PYTHONPATH
export PYTHONNOUSERSITE=1

2. I loaded the necessary modules

module load compiler/gcc/4.9 R/3.2 ruby/2.1

3. in ~/.bashrc file: I added

export PATH=$BCBIO_PATH/anaconda/bin:$BCBIO_PATH/bin:$PATH
export LD_LIBRARY_PATH=$BCBIO_PATH/lib:$LD_LIBRARY_PATH
export PERL5LIB=$BCBIO_PATH/lib/perl5:$BCBIO_PATH/lib/perl5/site_perl:$PERL5LIB

4. I installed the bcbio
 
wget https://raw.github.com/chapmanb/bcbio-nextgen/master/scripts/bcbio_nextgen_install.py

python bcbio_nextgen_install.py $BCBIO_PATH --tooldir=$BCBIO_PATH --isolate --genomes GRCh37 --aligners bwa --aligners bowtie2

5. I updated gatk

bcbio_nextgen.py upgrade --tools --toolplus gatk=$HOME/tmp/bin/GenomeAnalysisTK.jar

6. I updated data

bcbio_nextgen.py upgrade --tools --toolplus data --data

7. I tried to update cadd and used the following command. The error appeared

bcbio_nextgen.py upgrade --toolplus cadd
Traceback (most recent call last):
  File "/work/.../bcbio/anaconda/bin/bcbio_nextgen.py", line 35, in <module>
    from bcbio.distributed import runfn, clargs
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 11, in <module>
    from bcbio.distributed import multitasks
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 3, in <module>
    from bcbio import heterogeneity, structural, utils, chipseq, upload
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/__init__.py", line 10, in <module>
    from bcbio.heterogeneity import bubbletree, phylowgs, sciclone, theta
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/bubbletree.py", line 17, in <module>
    from bcbio.structural import regions
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 10, in <module>
    from bcbio.structural import (battenberg, cn_mops, cnvkit, delly, ensemble,
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/cn_mops.py", line 20, in <module>
    from bcbio.structural import shared as sshared
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/shared.py", line 15, in <module>
    from bcbio.bam import callable
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/bam/callable.py", line 29, in <module>
    from bcbio.distributed import multi, prun
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/prun.py", line 9, in <module>
    from bcbio.distributed import clusterk, ipython, multi, resources
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipython.py", line 26, in <module>
    from cluster_helper import cluster as ipython_cluster
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/cluster_helper/cluster.py", line 22, in <module>
    from ipyparallel import Client
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/__init__.py", line 14, in <module>
    from ipykernel.pickleutil import Reference, can_map
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>
    from .connect import *
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/connect.py", line 12, in <module>
    from IPython.core.profiledir import ProfileDir
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/__init__.py", line 45, in <module>
    from .config.loader import Config
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/config/__init__.py", line 6, in <module>
    from .application import *
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/config/application.py", line 19, in <module>
    from IPython.config.configurable import SingletonConfigurable
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/config/configurable.py", line 14, in <module>
    from IPython.utils.text import indent, wrap_paragraphs
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/text.py", line 28, in <module>
    from IPython.external.path import path
ImportError: cannot import name path

Can you help me to debug it? Thanks a lot!

**Comment by chapmanb** [comment]
Sorry about the problem. The new IPython contains a new directory structure and somehow it looks like the install doesn't have the right version. You should be able to fix with a manual installation of the latest version:
```
/work/.../bcbio/anaconda/bin/pip uninstall ipython
/work/.../bcbio/anaconda/bin/conda uninstall ipython
/work/.../bcbio/anaconda/bin/conda install ipython ipyparallel
```
Hope this gets it working again for you.

**Comment by pengxiao78** [comment]
I have done the three steps. However, there is a new error when I updated cadd.

bcbio_nextgen.py upgrade --toolplus cadd

/work/.../anaconda/lib/python2.7/site-packages/IPython/config.py:13: ShimWarning: The `IPython.config` package has been deprecated. You should import from traitlets.config instead.
  "You should import from traitlets.config instead.", ShimWarning)
Traceback (most recent call last):
  File "/work/.../bcbio/anaconda/bin/bcbio_nextgen.py", line 35, in <module>
    from bcbio.distributed import runfn, clargs
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 11, in <module>
    from bcbio.distributed import multitasks
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 3, in <module>
    from bcbio import heterogeneity, structural, utils, chipseq, upload
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/__init__.py", line 10, in <module>
    from bcbio.heterogeneity import bubbletree, phylowgs, sciclone, theta
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/bubbletree.py", line 17, in <module>
    from bcbio.structural import regions
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 10, in <module>
    from bcbio.structural import (battenberg, cn_mops, cnvkit, delly, ensemble,
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/cn_mops.py", line 20, in <module>
    from bcbio.structural import shared as sshared
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/shared.py", line 15, in <module>
    from bcbio.bam import callable
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/bam/callable.py", line 29, in <module>
    from bcbio.distributed import multi, prun
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/prun.py", line 9, in <module>
    from bcbio.distributed import clusterk, ipython, multi, resources
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipython.py", line 26, in <module>
    from cluster_helper import cluster as ipython_cluster
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/cluster_helper/cluster.py", line 22, in <module>
    from ipyparallel import Client
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/__init__.py", line 14, in <module>
    from ipykernel.pickleutil import Reference, can_map
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>
    from .connect import *
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/connect.py", line 12, in <module>
    from IPython.core.profiledir import ProfileDir
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/__init__.py", line 48, in <module>
    from .core.application import Application
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/application.py", line 24, in <module>
    from IPython.core import release, crashhandler
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/crashhandler.py", line 28, in <module>
    from IPython.core import ultratb
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/ultratb.py", line 116, in <module>
    from IPython.utils import path as util_path
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/path.py", line 21, in <module>
    from IPython.utils.process import system
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/process.py", line 28, in <module>
    from ._process_posix import _find_cmd, system, getoutput, arg_split, check_pid
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/_process_posix.py", line 24, in <module>
    from IPython.external import pexpect
ImportError: cannot import name pexpect

**Comment by chapmanb** [comment]
Sorry about the problems. I'm confused as to what happened but you still have the old IPython directory since that failing line should be `import pyexpect` in the 4.0 versions of IPython. You could try a more aggressive approach to remove the old versions:
```
/work/.../bcbio/anaconda/bin/pip uninstall ipython
/work/.../bcbio/anaconda/bin/conda uninstall ipython
rm -rf /work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython
/work/.../bcbio/anaconda/bin/conda install ipython ipyparallel
```
Sorry to not have better ideas but I'm not sure how it got in this state. Hopefully this fixes it for you.

**Comment by pengxiao78** [comment]
New error. Thank you for your help!

pip uninstall ipython

DEPRECATION: Uninstalling a distutils installed project (ipython) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
Uninstalling ipython-4.0.0:
  /work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/ipython-4.0.0-py2.7.egg-info
Proceed (y/n)? y
  Successfully uninstalled ipython-4.0.0
( **@login:/work/.../bcbio/anaconda/bin ) conda uninstall ipython
Fetching package metadata: ....

Package plan for package removal in environment /work/.../bcbio/anaconda:

The following packages will be REMOVED:

    ipython: 4.0.0-py27_0

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|########################################################################################################################################################| 100%
( **@login:/work/.../bcbio/anaconda/bin ) rm -rf /work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython
( **@login:/work/.../bcbio/anaconda/bin ) ls /work/.../anaconda/lib/python2.7/site-packages/ | grep 'IPython'
( **@login:/work/unmc_gudalab/pxiao/bcbio/anaconda/bin ) conda uninstall ipython ipyparallel
Fetching package metadata: ....

Package plan for package removal in environment /work/.../bcbio/anaconda:

The following packages will be REMOVED:

    ipyparallel: 4.0.2-py27_0

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|########################################################################################################################################################| 100%
( **@login:/work/.../bcbio/anaconda/bin ) cd ..
( **@login:/work/.../bcbio/anaconda ) cd ..
( **@login:/work/.../bcbio ) bcbio_nextgen.py upgrade --toolplus cadd
Traceback (most recent call last):
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/bin/bcbio_nextgen.py", line 35, in <module>
    from bcbio.distributed import runfn, clargs
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 11, in <module>
    from bcbio.distributed import multitasks
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 3, in <module>
    from bcbio import heterogeneity, structural, utils, chipseq, upload
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/__init__.py", line 10, in <module>
    from bcbio.heterogeneity import bubbletree, phylowgs, sciclone, theta
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/bubbletree.py", line 17, in <module>
    from bcbio.structural import regions
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 10, in <module>
    from bcbio.structural import (battenberg, cn_mops, cnvkit, delly, ensemble,
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/cn_mops.py", line 20, in <module>
    from bcbio.structural import shared as sshared
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/shared.py", line 15, in <module>
    from bcbio.bam import callable
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/bam/callable.py", line 29, in <module>
    from bcbio.distributed import multi, prun
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/prun.py", line 9, in <module>
    from bcbio.distributed import clusterk, ipython, multi, resources
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipython.py", line 26, in <module>
    from cluster_helper import cluster as ipython_cluster
  File "/work/unmc_gudalab/pxiao/bcbio/anaconda/lib/python2.7/site-packages/cluster_helper/cluster.py", line 22, in <module>
    from ipyparallel import Client
ImportError: No module named ipyparallel

**Comment by chapmanb** [comment]
The last command should be `conda install ipython ipyparallel`. You uninstalled IPython but didn't re-install it. Hope this helps.

**Comment by pengxiao78** [comment]
Sorry for the mistake. I noticed that and deleted the previous post. However, the following error came.

( *@login:/work/.../bcbio/anaconda/bin ) ls /work/.../bcbio/anaconda/lib/python2.7/site-packages/ | grep 'IPython'
( *@login:/work/.../bcbio/anaconda/bin ) conda install ipython ipyparallel
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /work/.../bcbio/anaconda:

The following NEW packages will be INSTALLED:

    ipyparallel: 4.0.2-py27_0
    ipython:     4.0.0-py27_0

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|########################################################################################################################################################| 100%
( *@login:/work/.../bcbio/anaconda/bin ) cd ..
( *@login:/work/.../bcbio/anaconda ) cd ..
( *@login:/work/.../bcbio ) bcbio_nextgen.py upgrade --toolplus cadd
/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/config.py:13: ShimWarning: The `IPython.config` package has been deprecated. You should import from traitlets.config instead.
  "You should import from traitlets.config instead.", ShimWarning)
Traceback (most recent call last):
  File "/work/.../bcbio/anaconda/bin/bcbio_nextgen.py", line 35, in <module>
    from bcbio.distributed import runfn, clargs
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 11, in <module>
    from bcbio.distributed import multitasks
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 3, in <module>
    from bcbio import heterogeneity, structural, utils, chipseq, upload
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/__init__.py", line 10, in <module>
    from bcbio.heterogeneity import bubbletree, phylowgs, sciclone, theta
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/heterogeneity/bubbletree.py", line 17, in <module>
    from bcbio.structural import regions
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/__init__.py", line 10, in <module>
    from bcbio.structural import (battenberg, cn_mops, cnvkit, delly, ensemble,
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/cn_mops.py", line 20, in <module>
    from bcbio.structural import shared as sshared
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/structural/shared.py", line 15, in <module>
    from bcbio.bam import callable
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/bam/callable.py", line 29, in <module>
    from bcbio.distributed import multi, prun
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/prun.py", line 9, in <module>
    from bcbio.distributed import clusterk, ipython, multi, resources
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipython.py", line 26, in <module>
    from cluster_helper import cluster as ipython_cluster
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/cluster_helper/cluster.py", line 22, in <module>
    from ipyparallel import Client
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipyparallel/__init__.py", line 14, in <module>
    from ipykernel.pickleutil import Reference, can_map
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>
    from .connect import *
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/ipykernel/connect.py", line 12, in <module>
    from IPython.core.profiledir import ProfileDir
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/__init__.py", line 48, in <module>
    from .core.application import Application
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/application.py", line 24, in <module>
    from IPython.core import release, crashhandler
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/crashhandler.py", line 28, in <module>
    from IPython.core import ultratb
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/core/ultratb.py", line 116, in <module>
    from IPython.utils import path as util_path
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/path.py", line 21, in <module>
    from IPython.utils.process import system
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/process.py", line 28, in <module>
    from ._process_posix import _find_cmd, system, getoutput, arg_split, check_pid
  File "/work/.../bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/_process_posix.py", line 24, in <module>
    from IPython.external import pexpect
ImportError: cannot import name pexpect

**Comment by chapmanb** [comment]
Sorry I'm really confused as to what is happening. The failing line, 24 from `anaconda/lib/python2.7/site-packages/IPython/utils/_process_posix.py`, in your trace should be just `import pexpect`, which is what I see if I install ipython 4.0.0 from conda. Is it possible you're installing and running from different directories and I'm missing seeing it with the obscured paths in your tracebacks? I'm not sure how else we're consistently seeing the old version.

Sorry to not have a better idea. Reading back through I think this was triggered through installing GEMINI as the updates to IPython 4.0 haven't gotten pushed to a release yet. I wonder if running `gemini update --devel` would help at all.

Hope this helps some.

**Comment by pengxiao78** [comment]
Hi Brad,

The installing and running are in the same directory. I tried gemini update --devel and still it doesn’t work. I noticed that when I type “ipython” in my command prompt, I got the error I sent to you earlier, which is

File "./bcbio/anaconda/lib/python2.7/site-packages/IPython/utils/_process_posix.py", line 24, in <module>
    from IPython.external import pexpect
ImportError: cannot import name pexpect

If I type “python” and import “pexpect” inside python and I can actually import that. And the module “pexpect” is laid under the bcbio directory here “./bcbio/anaconda/lib/python2.7/site-packages/pexpect”. In this case, the command in line 24 of “_process_posix.py” seems redundant. Therefore, I changed all the “from IPython.external import pexpect” to just “import pexpect” wherever it gives that problem. Now I am getting this error.

  File "/-----/bcbio/anaconda/lib/python2.7/site-packages/IPython/core/oinspect.py", line 31, in <module>
    from IPython.core import page
  File "/-----/bcbio/anaconda/lib/python2.7/site-packages/IPython/core/page.py", line 26, in <module>
    from IPython.core.display import display
  File "/-----/bcbio/anaconda/lib/python2.7/site-packages/IPython/core/display.py", line 15, in <module>
    from IPython.core.formatters import _safe_get_formatter_method
  File "/-----/bcbio/anaconda/lib/python2.7/site-packages/IPython/core/formatters.py", line 215, in <module>
    @decorator
TypeError: 'module' object is not callable

So, I assume that I cannot do this changes like this and it seems endless as well.

I also have one important point forgot to tell you.
The first time I have run these command to upgrade bcbio and it works just fine.  I mentioned these in the previous email.

bcbio_nextgen.py upgrade --tools --toolplus gatk=$HOME/tmp/bin/GenomeAnalysisTK.jar                        ##WORKS
bcbio_nextgen.py upgrade --tools --toolplus data --data               ## WORKS

bcbio_nextgen.py upgrade --toolplus cadd          ##Now the error came out

So the attempt to “run bcbio_nextgen.py upgrade --toolplus cadd ” has failed and suddenly, running previous two commands (GATK and data upgrade) are now giving the same error as “cadd”. It seems that the ipython crashed since I run the third step.  Thank you.

**Comment by chapmanb** [comment]
Thanks for the additional digging. My thinking is that this command:
```
bcbio_nextgen.py upgrade --tools --toolplus data --data
```
caused installation of IPython 3.x which broken bcbio, since it's now expecting IPython 4.x. Since that installs GEMINI, I suspected it reverted IPython to 3.x, or at least a partial install of 3.x.

All of the errors indicate you still have a 3.x version of IPython, or at least part of it, in `bcbio/anaconda/lib/python2.7/site-packages/IPython`. As you noted, the code should be `import pexpect` in 4.x, but the link you have is in 3.x. Manually changing things won't help since you apparently have a persistent 3.x install somewhere.

Some ideas for debugging:

- What does `bcbio/anaconda/bin/conda list ipy` report? If there are any 3.x versions we could remove them and try re-installing.
- What does `ls -lh bcbio/anaconda/lib/python2.7/site-packages/IPython` report? Does it have a `core/release.py` file that has 4.x version information in it?

Sorry to not have a better clue but hope this explains my confusion and where we're at.

**Comment by pengxiao78** [comment]
Thank you for your help! The first command bcbio/anaconda/bin/conda list ipy did not show any 3.x versions.

# packages in environment at ./bcbio/anaconda:
#
ipykernel                 4.0.3                    py27_0
ipyparallel               4.0.2                    py27_0
ipython                   4.0.0                    py27_0
ipython-cluster-helper    0.4.9                    py27_0
ipython-genutils          0.1.0                     <pip>
ipython_genutils          0.1.0                    py27_0
scipy                     0.16.0               np19py27_1

But I did find the version is 3 in the release.py file as follows

# IPython version information.  An empty _version_extra corresponds to a full
# release.  'dev' as a _version_extra string means this is a development
# version
_version_major = 3
_version_minor = 2
_version_patch = 1
# _version_extra = 'dev'
# _version_extra = 'rc1'
_version_extra = ''  # Uncomment this for full releases

# release.codename is deprecated in 2.0, will be removed in 3.0
codename = ''

Do you have further idea on how to handle this mixed version situation?

Thanks again!

**Comment by pengxiao78** [comment]
Hi,

Eventually,  the problem has been solved! The previous uninstallation by pip uninstall ipython was not complete and I removed ./bcbio/anaconda/bin/ipython . I ran pip install ipython and ipython 4.0 was installed successfully. Then, I could run bcbio_nextgen.py.  
Thanks a lot for helping me with debugging!

FYI.
( *@login:./bcbio ) pip uninstall ipython
DEPRECATION: Uninstalling a distutils installed project (ipython) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
Uninstalling ipython-4.0.0:
 ./bcbio/anaconda/lib/python2.7/site-packages/ipython-4.0.0-py2.7.egg-info
Proceed (y/n)? y
  Successfully uninstalled ipython-4.0.0
( *@login:./bcbio ) which ipython
./bcbio/anaconda/bin/ipython
( *@login:./bcbio ) rm ./bcbio/anaconda/bin/ipython
( *@login:./bcbio ) pip install ipython
Collecting ipython
  Downloading ipython-4.0.0-py2-none-any.whl (730kB)
    100% |████████████████████████████████| 733kB 486kB/s
Requirement already satisfied (use --upgrade to upgrade): decorator in ./anaconda/lib/python2.7/site-packages (from ipython)
Requirement already satisfied (use --upgrade to upgrade): simplegeneric>0.8 in ./anaconda/lib/python2.7/site-packages (from ipython)
Requirement already satisfied (use --upgrade to upgrade): pexpect in ./anaconda/lib/python2.7/site-packages (from ipython)
Requirement already satisfied (use --upgrade to upgrade): traitlets in ./anaconda/lib/python2.7/site-packages (from ipython)
Requirement already satisfied (use --upgrade to upgrade): pickleshare in ./anaconda/lib/python2.7/site-packages (from ipython)
Requirement already satisfied (use --upgrade to upgrade): path.py in ./anaconda/lib/python2.7/site-packages (from pickleshare->ipython)
Installing collected packages: ipython
Successfully installed ipython-4.0.0
